### ISSUE TITLE
commit git problem

### ISSUE THREAD

**Issue by erfansaheb** [spyder-ide/spyder#15914]
### What steps will reproduce the problem?

just right clicked on the file and chose commit and got this error

### Traceback
```python-traceback
Traceback (most recent call last):
  File "C:\Users\erfan\Anaconda3\lib\site-packages\spyder\plugins\explorer\widgets\explorer.py", line 403, in <lambda>
    triggered=lambda: self.vcs_command('commit'),
  File "C:\Users\erfan\Anaconda3\lib\site-packages\spyder\plugins\explorer\widgets\explorer.py", line 1448, in vcs_command
    vcs.run_vcs_tool(path, action)
  File "C:\Users\erfan\Anaconda3\lib\site-packages\spyder\utils\vcs.py", line 87, in run_vcs_tool
    programs.run_program(tool, args, cwd=path)
  File "C:\Users\erfan\Anaconda3\lib\site-packages\spyder\utils\programs.py", line 278, in run_program
    return subprocess.Popen(fullcmd, **subprocess_kwargs)
  File "C:\Users\erfan\Anaconda3\lib\subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Users\erfan\Anaconda3\lib\subprocess.py", line 1311, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
NotADirectoryError: [WinError 267] The directory name is invalid
```

## Versions

* Spyder version: 5.0.3 
* Python version: 3.8.8
* Qt version: 5.12.9
* PyQt5 version: 5.12.3
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  1.6.0 (OK)
cookiecutter >=1.6.0          :  1.7.2 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.6.0               :  7.22.0 (OK)
jedi =0.17.2                  :  0.17.2 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  23.0.1 (OK)
nbconvert >=4.0               :  6.0.7 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
paramiko >=2.4.0              :  2.7.2 (OK)
parso =0.7.0                  :  0.7.0 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.8.0 (OK)
pygments >=2.0                :  2.9.0 (OK)
pylint >=1.0                  :  2.8.3 (OK)
pyls >=0.36.2;<1.0.0          :  0.36.2 (OK)
pyls_black >=0.4.6            :  0.4.6 (OK)
pyls_spyder >=0.3.2;<0.4.0    :  0.3.2 (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.1.10 (OK)
qtawesome >=1.0.2             :  1.0.2 (OK)
qtconsole >=5.1.0             :  5.1.0 (OK)
qtpy >=1.5.0                  :  1.9.0 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=39.0.0           :  52.0.0.post20210125 (OK)
sphinx >=0.6.6                :  4.0.1 (OK)
spyder_kernels >=2.0.3;<2.1.0 :  2.0.3 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3;<2.0.0      :  1.0.2 (OK)
zmq >=17                      :  20.0.0 (OK)

# Optional:
cython >=0.21                 :  0.29.23 (OK)
matplotlib >=2.0.0            :  3.3.4 (OK)
numpy >=1.7                   :  1.20.2 (OK)
pandas >=1.1.1                :  1.2.4 (OK)
scipy >=0.17.0                :  1.6.2 (OK)
sympy >=0.7.3                 :  1.8 (OK)
```

**Comment by steff456** [comment]
Hi @erfansaheb,

Is this a recurrent error? Can you please upload a gif showing what your behavior so we can try to reproduce it on our end?

Thanks for reporting!

**Comment by ccordoba12** [comment]
Hey @steff456 and @erfansaheb, this is a duplicate of issue #15458 and I fixed it already.

The fix will be part of our 5.1.0 version, to be released at the end of July.
